### DISCUSSION TITLE
Very high CPU usage?

### DISCUSSION THREAD

**Discussion by jeremehancock** [luigi311/JellyPlex-Watched#105] [Q&A]
I just set this up for my Plex/Jellyfin server and it is great!!

It is working great, but I do see really high spikes in CPU when it runs. I am concerned that it will cause brief issues with playback. Is this a known "issue", or are there any ways to reduce the spikes?

I have mine setup to run every hour like the example `.env` has.

Thanks

**Comment by luigi311** [comment]
Interesting so I do have it setup to thread as many times as possible for the plex library so each plex library is handled in a new thread. Jellyfin is also setup to be async so that will go as fast as possible. The plex thread count is handled by whatever the os reports as cores *2 so in theory it will use as many as you got up to the limit of 32. I can add a variable so you can set an upper limit manually.

**Reply by jeremehancock** [reply]
Thank you for the follow-up. Yeah my server has 4 cores and I see them all jump to 95% or so while JellyPlex-Watched runs. It doesn't last long, but still would be nice to not have it completely max things out.

Thanks!

**Reply by jeremehancock** [reply]
@luigi311 Do I need to put in a feature request for being able to set the variable that you mentioned?

**Comment by luigi311** [comment]
Theres no a MAX_THREAD variable you can set in the dev branch, let me know if that works for you guys

**Reply by jeremehancock** [reply]
I just updated to dev using the dev tag in docker. I set MAX_THREAD = "2" to see if it would have an effect. Doesn't seem to change anything. I will say that the spikes are primarily with generating the Plex stuff.

Maybe I am not setting it up correctly?

Thanks so much

**Comment by Minxster** [comment]
I have a reasonably large library of "stuff", and I to noted it hit my CPU hard, though it did seem like a nice spread over the cores 🤷‍♂️. It's more the Plex side of things as the Jellyfin part seemed quite quick. Do you think we could set a 'nice' level so that we can run it as a very low priority? This would remove some of the concerns I've been having on the frequency of running this?

**Reply by luigi311** [reply]
hmm ive never messed with nice especially not inside a container. If you are using a container it might be easier to just cpu pin it if MAX_THREAD doesnt seem to be working. If you are using unraid you should be able to go to settings and use the cpu pinning settings, if not then you need to use something like --cpus or --cpuset-cpus

**Reply by Minxster** [reply]
I've got some 30 containers running on one VM, so setting cpu limits is possible. But I don't mind it using all the cores, to get the transfers done as quick as possible. Which is where nice would really help out 🙂
